### PR TITLE
wip: Use world.key instead of world.id to generate entity PDA

### PR DESCRIPTION
**Pull request by crypto-vincent**
| Status  | Type  | ⚠️ Core Change | Issue |
| :---: | :---: | :---: | :--: |
| Ready| Refactor | No | -- |

## Problem

When adding a new entity using seeds, we have to fetch the WorldId which can needs to be resolved from the onchain state. This means we have to do an RPC call for each transaction we are trying to prepare in those cases.

## Solution

An easy solution is just to allow users to generate the Entity PDA from the worldPda instead, that way the users avoid unecessary lookups from the on-chain state when using entity seeds.

## Notes

Also change the naming from "extraSeed" to "seed" in the entity API, since this seed effectively replace the "entity id", it's not an addition on top of it.

### CODE REVIEW

File: clients/bolt-sdk/src/world/transactions.ts [hunk 1]
```diff
@@ -70,19 +70,20 @@ export async function AddEntity({
...
 }): Promise<{ transaction: Transaction; entityPda: PublicKey }> {
-  const worldInstance = await World.fromAccountAddress(connection, world);
-  const worldId = new BN(worldInstance.id);
-  const entityPda =
-    seed !== undefined
-      ? FindEntityPda({ worldId, seed })
-      : FindEntityPda({ worldId, entityId: new BN(worldInstance.entities) });
+  let entityPda: PublicKey;
+  if (seed !== undefined) {
+    entityPda = FindEntityPda({ world, seed });
```
Comment (crypto-vincent) [inline clients/bolt-sdk/src/world/transactions.ts:75]: this is now the happy path, much faster to generate transactions

**Comment by notdanilo** [comment]
This PR seems old. Is it still an issue?

**Comment by crypto-vincent** [comment]
> This PR seems old. Is it still an issue?

Hello there, yes latest master still seem to have the issue: we always have to read the latest world.Id through a "World.fromAccountAddress" before creating any new entity (but seed-based entities may not need to do that in theory)

**Comment by notdanilo** [comment]
I am closing this PR in favor of #120 because I applied the same changes in there

**Pull request closed without merging**